docker/lr-cartographer/Dockerfile

```diff
@@ -0,0 +1,153 @@
+# Start with a good base python3 image:
+#FROM ubuntu:19.10
+FROM ubuntu:21.04
+MAINTAINER Jonn Smith <jonn@broadinstitute.org>
+
+# Make sure we don't need to interact with any package installations:
+ARG DEBIAN_FRONTEND=noninteractive
+
+## Install any needed packages specified in requirements.txt
+#RUN pip install --trusted-host pypi.python.org -r requirements.txt
+
+# Set the working directory to /
+WORKDIR /
+
+################################################################################
+# Install system packages:
+
+# Make sure we can actually talk to package repos:
+RUN apt-get update 
+
+# Development / prereqs for building software:
+RUN apt-get -y install make gcc g++ autoconf sudo 
+
+# Utilities / tools:
+RUN apt-get -y install git bash vim time bc sed perl wget curl bzip2 man 
+
+# Reqs for numpy/scipy:
+RUN apt-get -y install libc-dev zlib1g-dev gfortran 
+
+# Pysam remaining requirements:
+RUN apt-get -y install liblzma-dev libbz2-dev
+
+# Get libcurses:
+RUN apt-get -y install libncurses5-dev libncursesw5-dev
+
+# Get python 3.8:
+RUN apt-get install -y software-properties-common
+RUN apt-get install -y libssl-dev libncurses5-dev libsqlite3-dev libreadline-dev libgdm-dev libdb4o-cil-dev libpcap-dev libtk8.6
+RUN cd /opt && \
+		wget https://www.python.org/ftp/python/3.8.2/Python-3.8.2.tgz && \
+		tar -xf Python-3.8.2.tgz && \
+		cd Python-3.8.2 && \
+		./configure --enable-optimizations && make altinstall && \
+		python3.8 -m pip install --upgrade pip
+
+# Install required Python packages: 
+RUN python3.8 -m pip install cython
+RUN python3.8 -m pip install mappy
+RUN python3.8 -m pip install --no-cache-dir numpy
+RUN python3.8 -m pip install --no-cache-dir matplotlib
+RUN python3.8 -m pip install --no-cache-dir scipy
+RUN python3.8 -m pip install pysam
+RUN python3.8 -m pip install biopython
+
+# Link python 3.8 to the global python executable:
+RUN ln -s /usr/local/bin/python3.8 /usr/local/bin/python
+
+################################################################################
+# Install local settings:
+RUN echo "set nowrap\nset number\nset hlsearch" > ~/.vimrc
+
+################################################################################
+# Install utils:
+
+COPY tesserae-1.2.0.tar.gz /
+RUN python3.8 -m pip install /tesserae-1.2.0.tar.gz
+RUN rm /tesserae-1.2.0.tar.gz 
+
+RUN git clone --recursive https://github.com/jonn-smith/seq-align.git
```

Comment:
why these un-official repos?
if a specific version is needed, why not pin it down to a release or a hash?

Reply:
I didn't need to do this for `seq-align` so I reverted it.

I have patches for both `MosaicAligner` and `TALON` that needed to be put in to use them.  I think they've been pushed up to the parent repos, but I'm not sure.